templates/release_checklist.md

```diff
@@ -0,0 +1,149 @@
+<!-- Checklist generated by release toolkit -->
+
+<!-- scheduled_release_only -->
+
+## Pre release
+
+- [ ] Clone or update the [release toolkit](https://href.li/?https://github.com/wordpress-mobile/release-toolkit-gutenberg-mobile) and review the [release script instructions](https://github.com/wordpress-mobile/release-toolkit-gutenberg-mobile/blob/develop/Releasing.md).
```

Comment:
I found it odd that it was in the "Create the release" section since it's mostly a prep step. For someone new it would be better to review the process a few days before. Or if it's been awhile since wrangling a release.
I suppose there could be an important change to the release script between, say Tuesday and Thursday but I doubt that would happen. In that case we could have the release automation script check for updated when it gets ran on Thursday